File: 6__Python__/pset6/readability/readability.py
text = input("Text: ")
# 0.0588 * L - 0.296 * S - 15.8
letters, words, sentences = 0, 1, 0
for char in text:
    if char.isalpha():
        letters += 1
    elif char.isspace():
        words += 1
    elif char in ['.', '!', '?']:
        sentences += 1

L = letters / words * 100
S = sentences / words * 100
index = 0.0588 * L - 0.296 * S - 15.8
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")

File: 2__Arrays__/pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // index = 0.0588 * L - 0.296 * S - 15.8
    // L = letters per 100 words, S = sentances per 100 words

    // Prompt user for input string
    string input = get_string("Text: ");

    // Counter letters, words, and sentences
    int letters = 0;
    int words = 1;
    int sentences = 0;
    for (int i = 0; input[i] != 0; i++)
    {
        if (isalpha(input[i]))
        {
            letters++;
        }
        if (isspace(input[i]))
        {
            words++;
        }
        if (input[i] == '.' || input[i] == '!' || input[i] == '?')
        {
            sentences++;
        }
    }

    // Output grade level
    // printf("%i letter(s)\n", letters);
    // printf("%i words(s)\n", words);
    // printf("%i sentence(s)\n", sentences);

    float L = 100 * letters / words;
    float S = 100 * sentences / words;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        int num = round(index);
        printf("Grade %i\n", num);
    }
}